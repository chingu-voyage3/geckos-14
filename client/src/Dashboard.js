import React, { Component } from 'react';

class Dashboard extends Component {
  render() {
    return (
      <div>Dashboard Page</div>
    );
  }
}

export default Dashboard;
