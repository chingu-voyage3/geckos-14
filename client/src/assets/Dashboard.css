.dashboard {
  display: grid;
  grid-template-columns: 2fr 6fr;
  margin: 20px 20px 20px 20px;
}
.thing-panel {
  display: grid;
  grid-template-rows: 1fr 9fr;
  border: solid 1px;
}
.viz-panel {
  display: grid;
  grid-template-rows: 1fr 9fr;
  border: solid 1px;
}

.control-panel {
  display: grid;
  grid-auto-rows: 1fr 4fr;
  border: dashed 1px;
}
.control-panel-params {
  margin: 5px 20px 5px 20px;
  display: grid;
  grid-auto-rows: 20px;
}
.viz-control-panel-params {
  margin: 5px 20px 5px 20px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-auto-rows: 20px;
  grid-column-gap: 50px;
  grid-row-gap: 1em;
}
.control-panel-actions {
  display: grid;
  grid-template-columns: repeat(5, 40px);
  border: dashed 1px red;
}

.control-panel-action {
  border-radius: 5%;
  border: dashed 1px blue;
}
.control-panel-select {
  width: auto;
}
.control-panel-param {
  border: dashed 1px green;
}

.thing-list {
}
.thing {
  margin-left: 20px;
  margin-right: 20px;
  border: dashed 1px;
}
.thing-property {
  margin-left: 20px;
  margin-right: 20px;
  border: dashed 1px orange;
}
.thing-action {
  margin-left: 20px;
  margin-right: 20px;
  border: dashed 1px blue;
}
.thing-description {
  font-size: 0.7em;
}

.viz-grid {
  display: grid;
  grid-gap: 5px;
  grid-template-columns: repeat(auto-fit, 300px);
  grid-template-rows: repeat(auto-fit, 200px);
}
.viz {
  height: 200px;
  width: 300px;
  border: dashed 1px;
}
.viz-property {
}
.viz-action {
  padding: 20px 20px 20px 20px;
}
.viz-action label {
  padding-right: 20px;
}

.visible {
}

.hidden {
  display: none;
}
.switch {
  display: grid;
  grid-template-rows: repeat(2, 40px);
  border: dashed 1px red;
}
.switch-label {
  display: grid;
  grid-template-columns: repeat(3, 60px);
  border: dashed 1px blue;
  text-align: center;
}
/* The switch - the box around the slider */
.switch-box {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 34px;
}

/* Hide default HTML checkbox */
.switch-box input {
  display: none;
}

/* The slider */
.switch-slider {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.switch-slider:before {
  position: absolute;
  content: '';
  height: 26px;
  width: 26px;
  left: 4px;
  bottom: 4px;
  background-color: white;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

input:checked + .slider {
  background-color: #2196f3;
}

input:focus + .slider {
  box-shadow: 0 0 1px #2196f3;
}

input:checked + .slider:before {
  -webkit-transform: translateX(26px);
  -ms-transform: translateX(26px);
  transform: translateX(26px);
}

/* Rounded sliders */
.switch-slider.round {
  border-radius: 34px;
}

.switch-slider.round:before {
  border-radius: 50%;
}

.switch-info-true {
  color: green;
}
.switch-info-false {
  color: red;
}
.switch-name {
  text-align: center;
}
.switch-timestamp {
  text-align: center;
}
