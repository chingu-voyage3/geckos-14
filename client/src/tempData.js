export const lineData = [
  // { date: 12, temp: 25.5 },
  // { date: 13, temp: 25.6 },
  // { date: 14, temp: 24.3 },
  // { date: 15, temp: 25.8 }
];
export const barData = [
  { date: '12/12', hum: 95 },
  { date: '13/12', hum: 93 },
  { date: '14/12', hum: 98 },
  { date: '15/12', hum: 85 }
];
export const uiThemes = [
  { id: 'TH0', value: 'default', display: 'Select Viz Theme' },
  { id: 'TH1', value: 'GrayScale', display: 'GrayScale' },
  { id: 'TH2', value: 'Material', display: 'Material' }
];

export const modelTypes = [
  { id: 'MT0', value: 'default', display: 'Select Model Type' },
  { id: 'MT1', value: 'VictoryLine', display: 'VictoryLine' },
  { id: 'MT2', value: 'VictoryBar', display: 'VictoryBar' },
  { id: 'MT3', value: 'Status', display: 'Status' },
  { id: 'MT4', value: 'Display', display: 'Display' }
];

export const things = [
  {
    id: 'http://gateway.webofthings.io/',
    name: 'LondonTemp',
    description: 'Testing Thing',
    properties: {
      prop1: {
        name: 'prop1'
      },
      prop2: {
        name: 'prop2'
      }
    },
    actions: {
      act1: {
        name: 'act1'
      },
      act2: {
        name: 'act2'
      }
    }
  }
];
export const vizs = [
  {
    id: 'viz0',
    name: 'Temperature Sensor',
    thing_id: 'dev0',
    model: 'VictoryLine',
    x: 'date',
    y: 'temp',
    data: lineData,
    selected: false,
    design: 'GrayScale'
  },
  {
    id: 'viz1',
    name: 'Humidity Sensor',
    thing_id: 'dev1',
    model: 'VictoryBar',
    x: 'date',
    y: 'hum',
    data: barData,
    selected: false,
    design: 'Material'
  }
];

export const thingParams = [
  {
    label: 'Name :',
    type: 'input',
    name: 'name',
    options: []
  },
  {
    label: 'Url :',
    type: 'input',
    name: 'id',
    options: []
  }
];

export const vizParams = [
  {
    label: 'Name :',
    name: 'name',
    type: 'input',
    options: []
  },
  {
    label: 'Model :',
    name: 'model',
    type: 'select',
    options: modelTypes
  },
  {
    label: 'Source Thing:',
    name: 'thing_id',
    type: 'select',
    options: [{ id: 'SD0', value: 'default', display: 'Select Thing Source' }]
  },
  {
    label: 'Theme:',
    name: 'design',
    type: 'select',
    options: uiThemes
  }
];

export const testData = [
  { date: '12/12', hum: 22 },
  { date: '13/12', hum: 13 },
  { date: '14/12', hum: 5 },
  { date: '15/12', hum: 45 }
];

export const gridLayout = [
  {
    i: 'Temp Line',
    x: 0,
    y: 0,
    w: 1,
    h: 1
  },
  {
    i: 'Humidity Bar',
    x: 1,
    y: 0,
    w: 1,
    h: 1
  }
];

export const getStartedContent = [
  {
    id: 'step1',
    title: 'Context',
    text:
      'We are using the concept of "Things" described in the "Building the Web of Things" book written by Dominique D. Guinard and Vlad M. Trifa: A Thing with an uppercase "T" refers to a smart thing. A physical object that\'s digitally augmented with one or more of the following: Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow.'
  },
  {
    id: 'step2',
    title: 'Requirements',
    text:
      'Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow.'
  },
  {
    id: 'step3',
    title: 'Connect Things',
    text:
      'Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow.'
  },
  {
    id: 'step4',
    title: 'Display Things',
    text:
      'Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow. Unicorn hacker waterfall is so 2000 and late parallax grok piverate responsive affordances fund intuitive. Cortado bootstrapping personas ideate SpaceTeam user story co-working paradigm integrate entrepreneur 360 campaign pair programming. Human-centered design user story unicorn disrupt actionable insight pair programming prototype driven. Integrate entrepreneur fund co-working thought leader human-centered design co-working engaging. Minimum viable product user story parallax agile latte agile hacker workflow experiential physical computing engaging personas long shadow.'
  }
];

export const devPanelActions = [
  {
    name: 'Discover',
    onSelect: false
  },
  {
    name: 'Edit',
    onSelect: true
  },
  {
    name: 'Del',
    onSelect: true
  }
];
export const vizPanelActions = [
  {
    name: 'Add',
    onSelect: false
  },
  {
    name: 'Edit',
    onSelect: true
  },
  {
    name: 'Del',
    onSelect: true
  }
];
export const piGateway = {
  data: {
    message: 'testing root url',
    links: {
      properties: {
        temp: {
          name: 'Temperature Sensor',
          description: 'An ambient temperature sensor.',
          values: {
            t: {
              name: 'Temperature sensor',
              description: 'The temperature in celsius',
              unit: 'celsius',
              customFields: {
                gpio: 12
              }
            }
          },
          tags: ['sensor', 'public', 'indoors'],
          data: [
            {
              t: 36,
              timestamp: '2018-01-04T15:44:05.589Z'
            },
            {
              t: 39,
              timestamp: '2018-01-04T15:44:10.593Z'
            }
          ]
        }
      },
      actions: { ledState: 'leds' }
    }
  },
  headers: { Link: '/model' }
};

export const piModel = {
  data: {
    message: 'testing model url',
    links: {
      product: {
        link: 'https://www.raspberrypi.org/products/raspberry-pi-2-model-b/',
        title: 'Product this Web Thing is based on'
      },
      properties: {
        link: '/properties',
        title: 'List of Properties',
        resources: {
          temperature: {
            name: 'Temperature Sensor',
            description: 'An ambient temperature sensor.',
            values: {
              t: {
                name: 'Temperature sensor',
                description: 'The temperature in celsius',
                unit: 'celsius',
                customFields: {
                  gpio: 12
                }
              }
            },
            tags: ['sensor', 'public', 'indoors'],
            data: [
              {
                t: 21,
                timestamp: '2018-01-05T20:07:16.096Z'
              },
              {
                t: 14,
                timestamp: '2018-01-05T20:07:21.102Z'
              }
            ]
          },
          humidity: {
            name: 'Humidity Sensor',
            description: 'An ambient humidity sensor.',
            values: {
              h: {
                name: 'Humidity',
                description: 'Percentage of Humidity',
                unit: '%',
                customFields: {
                  gpio: 12
                }
              }
            },
            tags: ['sensor', 'public'],
            data: [
              {
                h: 24,
                timestamp: '2018-01-05T20:07:16.096Z'
              },
              {
                h: 93,
                timestamp: '2018-01-05T20:07:21.102Z'
              },
              {
                h: 63,
                timestamp: '2018-01-05T20:07:26.108Z'
              }
            ]
          }
        }
      }
    }
  },
  headers: { Link: '/model' }
};

export const modelPrint = ['id', 'name', 'description', 'tags', 'customFields', 'links'];
export const thingPrint = [
  'name',
  'description',
  'type',
  'unit',
  'value',
  'frequency',
  'timestamp'
];
