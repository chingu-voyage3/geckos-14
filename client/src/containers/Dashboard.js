/** Dashboard Container :
The Dahshboard contains the state allowing to display the Device list,
 the Viz Grid and the control panels
 For now the data is pulled from a localFile.
 Later part of the data will be stored in cache and DB
 */

import React, { Component } from 'react';
import DevicePanel from '../components/dashboard/DevicePanel';
import VizPanel from '../components/dashboard/VizPanel';
import '../assets/Dashboard.css';
import * as d from '../tempData.js';

const tempSocket = new WebSocket('ws://devices.webofthings.io/pi/sensors/temperature');

class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      devices: d.devices,
      deviceParams: d.deviceParams,
      selectedDevice: {},
      vizs: d.vizs,
      vizParams: d.vizParams,
      selectedViz: {}
    };
  }
  editDevice = (id, newProps) => {
    // TODO: Add function that updates a Device
  };
  addDevice = device => {
    console.log('deviceAdded');
    this.setState({
      devices: [...this.state.devices, device]
    });
  };
  delDevice = id => {
    // TODO: Add function that deletes a Device
  };
  addViz = viz => {
    // console.log('viz to be Added', viz);
    // console.log('dash vizs state', this.state);
    this.setState({
      vizs: [...this.state.vizs, viz]
    });
  };
  editViz = (id, newProps) => {
    // TODO: Add function that updates a Viz
  };
  delDevice = id => {
    // TODO: Add function that deletes a Device
  };
  toogleSelectedDevice = id => {
    let tempDevices = this.state.devices;
    for (var i = 0; i < tempDevices.length; i++) {
      if (tempDevices[i].id === id) {
        this.setState({
          selectedDevice: tempDevices[i]
        });
      }
    }
  };
  toogleSelectedViz = id => {
    let tempVizs = this.state.vizs;
    for (var i = 0; i < tempVizs.length; i++) {
      if (tempVizs[i].id === id) {
        this.setState({
          selectedViz: tempVizs[i]
        });
      }
    }
  };

  addDataPoint = dataPoint => {
    let tempViz = this.state.vizs;
    for (var i = 0; i < tempViz.length; i++) {
      if (tempViz[i].name === dataPoint.name) {
        tempViz[i].data.push({
          date: tempViz[0].data.length,
          temp: dataPoint.value
        });
      }
    }
    this.setState({
      vizs: tempViz
    });
  };
  componentDidMount() {
    // tempSocket.onmessage = event => {
    //   const result = JSON.parse(event.data);
    //   this.addDataPoint(result);
    // };
  }

  render() {
    const actions = {
      addDevice: this.addDevice,
      addViz: this.addViz,
      toogleSelectedDevice: this.toogleSelectedDevice,
      toogleSelectedViz: this.toogleSelectedViz
    };
    return (
      <div className="dashboard">
        <DevicePanel
          devices={this.state.devices}
          actions={actions}
          params={this.state.deviceParams}
          selected={this.state.selectedDevice}
        />
        <VizPanel
          vizs={this.state.vizs}
          actions={actions}
          params={this.state.vizParams}
          selected={this.state.selectedViz}
        />
      </div>
    );
  }
}

export default Dashboard;
