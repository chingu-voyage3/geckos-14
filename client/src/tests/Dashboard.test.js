import React from 'react';
import Dashboard from '../containers/Dashboard';
import * as d from '../tempData';
import { shallow } from 'enzyme';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// This sets the mock adapter on the default instance
let mock = new MockAdapter(axios);

// Mock any GET request to /url
// arguments for reply are (status, data, headers)
mock.onGet('http://devices.webofthings.io/pi/sensors').reply(200, d.piSensors);
mock.onGet('http://devices.webofthings.io/pi/actuators').reply(200, d.piSensors);

it('renders without crashing', async () => {
  const wrapper = await shallow(<Dashboard />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
