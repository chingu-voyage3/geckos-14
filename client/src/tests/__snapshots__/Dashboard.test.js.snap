// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="dashboard"
>
  <DevicePanel
    actions={
      Object {
        "addDevice": [Function],
        "addViz": [Function],
        "toogleSelected": [Function],
      }
    }
    devices={
      Array [
        Object {
          "id": "1",
          "name": "LondonTemp",
          "selected": false,
          "type": "Thermometer",
          "url": "http://devices.webofthings.io/pi/sensors/temperature/",
        },
        Object {
          "id": "2",
          "name": "LondonHumidity",
          "selected": false,
          "type": "Humidity",
          "url": "http://devices.webofthings.io/pi/sensors/humidity/",
        },
      ]
    }
    params={
      Array [
        Object {
          "label": "Name :",
          "name": "name",
        },
        Object {
          "label": "Type :",
          "name": "type",
        },
        Object {
          "label": "Url :",
          "name": "url",
        },
      ]
    }
    selected={Object {}}
  />
  <VizPanel
    actions={
      Object {
        "addDevice": [Function],
        "addViz": [Function],
        "toogleSelected": [Function],
      }
    }
    params={
      Array [
        Object {
          "label": "Name :",
          "name": "name",
          "options": "",
        },
        Object {
          "label": "Model :",
          "name": "model",
          "options": Array [
            "LineChart",
            "BarChart",
            "Status",
            "Display",
          ],
        },
        Object {
          "label": "Device :",
          "name": "device_name",
          "options": "",
        },
      ]
    }
    selected={Object {}}
    vizs={
      Array [
        Object {
          "data": Array [],
          "deviceId": "1",
          "id": "1",
          "model": "VictoryLine",
          "name": "Temperature Sensor",
          "selected": false,
          "x": "date",
          "y": "temp",
        },
        Object {
          "data": Array [
            Object {
              "date": "12/12",
              "hum": 95,
            },
            Object {
              "date": "13/12",
              "hum": 93,
            },
            Object {
              "date": "14/12",
              "hum": 98,
            },
            Object {
              "date": "15/12",
              "hum": 85,
            },
          ],
          "deviceId": "2",
          "id": "2",
          "model": "VictoryBar",
          "name": "Humidity Sensor",
          "selected": false,
          "x": "date",
          "y": "hum",
        },
      ]
    }
  />
</div>
`;
